common/src/main/java/org/apache/uniffle/common/ShufflePartitionedData.java

```diff
@@ -21,12 +21,20 @@
 
 public class ShufflePartitionedData {
 
+  private static final ShufflePartitionedBlock[] EMPTY_BLOCK_LIST =
+      new ShufflePartitionedBlock[] {};
   private int partitionId;
-  private ShufflePartitionedBlock[] blockList;
+  private final ShufflePartitionedBlock[] blockList;
+  private final long totalBlockSize;
 
   public ShufflePartitionedData(int partitionId, ShufflePartitionedBlock[] blockList) {
     this.partitionId = partitionId;
-    this.blockList = blockList;
+    this.blockList = blockList == null ? EMPTY_BLOCK_LIST : blockList;
+    long size = 0L;
```

Comment:
This loop to summary the size could be operated in the `toPartitionedBlock` method, which could be avoid duplicate loop.

Reply:
@zuston Sorry for missing reply this comment!

I add a new constructor for production code, and left this original overloaded constructor for test code for consistency.